source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -340,6 +342,12 @@ Http::FilterDataStatus JsonTranscoderFilter::encodeData(Buffer::Instance& data,
     return Http::FilterDataStatus::Continue;
   }
 
+  // TODO(dio): Add support for streaming case.
+  if (has_http_body_output_) {
+    buildResponseFromHttpBodyOutput(*response_headers_, data);
```

Comment:
Are we assuming here that the entire HTTP response proto is available in a single `encodeData` invocation? I don't think that's true, even for the non-streaming case.

Reply:
Thanks, @htuch. It makes sense, it seems it needs to buffer the data. Will try to explore it.

Reply:
I think if we don't have enough data to decode for a single frame, we buffer it in here: https://github.com/envoyproxy/envoy/pull/3793/files#diff-4d3d9222b440350a28a7fc3775efd8fdR429.

Reply:
Yeah, fair enough. How do we actually continue processing when we're done though? Looks like we're just always returning Http::FilterDataStatus::StopIterationAndBuffer?

Reply:
@htuch I think @dio already does this correctly, `decoder_.decode(data, frames);` buffers the data, and it returns `Http::FilterDataStatus::StopIterationAndBuffer` to stop the iteration, because the filter need content-type from grpc-frame, and potentially grpc-status in trailer. The transcoder will stop filter iteration until it can send full response (i.e. until encodeTrailer) for unary cases.

Reply:
@dio can you add some test case to cover this? (one grpc frame split in two encodeData call)

Reply:
@lizan got it, makes sense, thanks.

Reply:
@lizan added.